### PR TITLE
[Shader Graph][HDRP] Fix Normals in Decal Master Node

### PR DESCRIPTION
**Pull request by alindmanUnity**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing? 
Decals in HDRP were not using reprojected inputs for Positions or Normals. 
Previous implementation of Normal From Height could produce NaN output. 

Previous behavior: 
DecalNormalFromHeight

Fixed behavior:
DecalNormalFromHeightFix

---
### Testing status

**Manual Tests**: What did you do?
- Other: Ran graphics tests locally, all still passed. 
Created a test scene and used the Normal From Height node on a Decal Master node, observed the fixed behavior (seen in gif above)
Confirmed that the change did not break any other set ups for decal master node. 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
Nothing yet, HDRP QA please advise on how to best add test coverage for this case. 

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
The default value for normal strength should be 0.01, not 0.001

**Review by alelievr — APPROVED** [review]

**Review by JMargevics — APPROVED** [review]
Checked decal SG on projector and as a mesh decal. No issues were found, normals look good ✅
image

Would be good to rebase on staging to get the fix for decals not working without Metal & AO enabled
( https://github.com/Unity-Technologies/Graphics/pull/195 )

**Comment by sebastienlagarde** [comment]
staging is coming to master very soon :)

File: com.unity.shadergraph/Editor/Data/Nodes/Artistic/Normal/NormalFromHeightNode.cs [hunk 1]
```diff
@@ -58,40 +60,46 @@ string GetFunctionName()
...
+                    s.AppendLine("void {0}({1} In, {2} Strength, $precision3 Position, $precision3x3 TangentMatrix, out {3} Out)",
                         GetFunctionName(),
                         FindInputSlot<MaterialSlot>(InputSlotId).concreteValueType.ToShaderString(),
+                        FindInputSlot<MaterialSlot>(StrengthSlotId).concreteValueType.ToShaderString(),
                         FindOutputSlot<MaterialSlot>(OutputSlotId).concreteValueType.ToShaderString());
                     using (s.BlockScope())
                     {
-                        s.AppendLine("$precision3 worldDirivativeX = ddx(Position * 100);");
-                        s.AppendLine("$precision3 worldDirivativeY = ddy(Position * 100);");
+                        s.AppendLine("$precision3 worldDirivativeX = ddx(Position);");
```
Comment (sebastienlagarde) [inline com.unity.shadergraph/Editor/Data/Nodes/Artistic/Normal/NormalFromHeightNode.cs:90]: there is a typo is the name worldDirivativeX  ? should be worldDerivativeX  ?

**Review by ghost — APPROVED** [review]
No visual differences for Normal from Height Node.
Normal from Height node confirmed for significant improvement in decal shaders.
Fixes to normal vector, position vector, and transformation spaces are confirmed and are significant improvements. While this requires some tricky math at the moment, it now enables significant improvements in use of decals with normals:
normaldecalfix2

File: com.unity.shadergraph/Editor/Data/Nodes/Artistic/Normal/NormalFromHeightNode.cs (hunk 1 above)
Comment (alindmanUnity) [reply com.unity.shadergraph/Editor/Data/Nodes/Artistic/Normal/NormalFromHeightNode.cs]: good catch! fixed

**Review by marctem — APPROVED** [review]

**Pull request merged**